jetty-util/src/test/java/org/eclipse/jetty/util/resource/ResourceAddPathTest.java

```diff
@@ -0,0 +1,231 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.eclipse.jetty.toolchain.test.FS;
+import org.eclipse.jetty.toolchain.test.MavenTestingUtils;
+import org.eclipse.jetty.util.TypeUtil;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ResourceAddPathTest
```

Comment:
I'd much rather test cases be added to the existing `ResourceTest` rather than create a special purpose unit test.    By creating multiple special purpose test that overlap with other test cases it just becomes too complex to see what is tested and then gaps emerge.   We have already gone down this path to some extent as we have `ResourceTest`, `FileSystemResourceTest`, `PathResourceTest`, `ResourceAliasTest`.  

We need to simplify and rationalize the testing rather than duplicate tests, so that `ResourceTest` does all the tests it can over all the different implementations of `Resource`.  Then we can have a `PathResourceTest` iff there are some tests that can only be done on PathResources and not other resources in general.    We should strive to get rid of  `ResourceAliasTest`, `FileSystemResourceTest` and definitely not add `ResourceAddPathTest`